site/app/models/gradeable/Gradeable.php

```diff
@@ -991,8 +901,18 @@ public function getDates() {
     }
 
     public function getStringThreadIds() {
-        return $this->isDiscussionBased() && is_array(json_decode($this->getDiscussionThreadId()))
-            ? implode(',', json_decode($this->getDiscussionThreadId())) : '';
+        return $this->isDiscussionBased() ? implode(',', $this->getDiscussionThreadId()) : '';
+    }
+
+    /**
+     * Gets a string of all of the hidden files for this gradeable
+     * @return string|null
```

Comment:
PHPDoc annotations are unnecessary if they don't specify anything more than the native PHP type annotation does.